src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -0,0 +1,603 @@
+    return super(RscCompile, cls).implementation_version() + [('RscCompile', 7)]
+
+  @classmethod
+  def register_options(cls, register):
+    super(RscCompile, cls).register_options(register)
+
+    rsc_outline_version = '0.0.0-294-d7114447'
+    scalameta_version = '4.0.0-M10'
+
+    # TODO: it would be better to have a less adhoc approach to handling
+    #       optional dependencies. See: https://github.com/pantsbuild/pants/issues/6390
+    cls.register_jvm_tool(
+      register,
+      'workaround-metacp-dependency-classpath',
+      classpath=[
+        JarDependency(org = 'org.scala-lang', name = 'scala-compiler', rev = '2.11.12'),
+        JarDependency(org = 'org.scala-lang', name = 'scala-library', rev = '2.11.12'),
+        JarDependency(org = 'org.scala-lang', name = 'scala-reflect', rev = '2.11.12'),
+        JarDependency(org = 'org.scala-lang.modules', name = 'scala-partest_2.11', rev = '1.0.18'),
+        JarDependency(org = 'jline', name = 'jline', rev = '2.14.6'),
+        JarDependency(org = 'org.apache.commons', name = 'commons-lang3', rev = '3.3.2'),
+        JarDependency(org = 'org.apache.ant', name = 'ant', rev = '1.8.2'),
+        JarDependency(org = 'org.pegdown', name = 'pegdown', rev = '1.4.2'),
+        JarDependency(org = 'org.testng', name = 'testng', rev = '6.8.7'),
+        JarDependency(org = 'org.scalacheck', name = 'scalacheck_2.11', rev = '1.13.1'),
+        JarDependency(org = 'org.jmock', name = 'jmock-legacy', rev = '2.5.1'),
+        JarDependency(org = 'org.easymock', name = 'easymockclassextension', rev = '3.1'),
+        JarDependency(org = 'org.seleniumhq.selenium', name = 'selenium-java', rev = '2.35.0'),
+      ],
+      custom_rules=[
+        Shader.exclude_package('*', recursive=True),]
+    )
+    cls.register_jvm_tool(
+      register,
+      'rsc_outline',
+      classpath=[
+          JarDependency(
+              org='com.twitter',
+              name='rsc_2.11',
+              rev=rsc_outline_version,
+          ),
+          JarDependency(
+              org='com.twitter',
+              name='mjar_2.11',
+              rev=rsc_outline_version,
+          ),
+          JarDependency(
+            org='org.scalameta',
+            name='metacp_2.11',
+            rev=scalameta_version,
+          ),
+          JarDependency(
+            org='org.scalameta',
+            name='metai_2.11',
+            rev=scalameta_version,
+          ),
+      ],
+      custom_rules=[
+        Shader.exclude_package('scala', recursive=True),
+        Shader.exclude_package('rsc', recursive=True),
+      ]
+      )
+
+  def register_extra_products_from_contexts(self, targets, compile_contexts):
+    super(RscCompile, self).register_extra_products_from_contexts(targets, compile_contexts)
+    # TODO when digests are added, if the target is valid,
+    # the digest should be loaded in from the cc somehow.
+    for target in targets:
+      rsc_outline_cc, compile_cc = compile_contexts[target]
```

Comment:
How about renaming `rsc_outline_` thingies as follows?
  * `rsc_outline_cc` to `rsc_cc`
  * `rsc_outline_classpath` to `rsc_classpath`
  * `rsc_outline_key_for_target` to `rsc_key_for_target`
  * `rsc_outline({})` to `rsc({})`
  * `rsc_outline_jobs` to `rsc_jobs`

Alternatively, if the convention is to say `zinc_compile_cc`, etc, let's do `rsc_compile_cc`, `rsc_compile_classpath`, etc.

Reply:
The `rsc_outline` prefixes came from a search and replace. I think the shorter naming scheme you are suggesting is fine.